cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -45,28 +45,36 @@ class parquet_field {
 std::string field_type_string(FieldType type)
 {
   switch (type) {
-    case ST_FLD_TRUE: return "bool(true)";
-    case ST_FLD_FALSE: return "bool(false)";
-    case ST_FLD_BYTE: return "int8";
-    case ST_FLD_I16: return "int16";
-    case ST_FLD_I32: return "int32";
-    case ST_FLD_I64: return "int64";
-    case ST_FLD_DOUBLE: return "double";
-    case ST_FLD_BINARY: return "binary";
-    case ST_FLD_STRUCT: return "struct";
-    case ST_FLD_LIST: return "list";
-    case ST_FLD_SET: return "set";
-    default: return "unknown(" + std::to_string(type) + ")";
+    case FieldType::BOOLEAN_TRUE: return "bool(true)";
+    case FieldType::BOOLEAN_FALSE: return "bool(false)";
+    case FieldType::I8: return "int8";
+    case FieldType::I16: return "int16";
+    case FieldType::I32: return "int32";
+    case FieldType::I64: return "int64";
+    case FieldType::DOUBLE: return "double";
+    case FieldType::BINARY: return "binary";
+    case FieldType::STRUCT: return "struct";
+    case FieldType::LIST: return "list";
+    case FieldType::SET: return "set";
+    default: return "unknown(" + std::to_string(static_cast<uint8_t>(type)) + ")";
   }
 }
 
 void assert_field_type(int type, FieldType expected)
 {
-  CUDF_EXPECTS(type == expected,
+  CUDF_EXPECTS(type == static_cast<int>(expected),
                "expected " + field_type_string(expected) + " field, got " +
                  field_type_string(static_cast<FieldType>(type)) + " field instead");
 }
 
+void assert_bool_field_type(int type)
+{
+  CUDF_EXPECTS(type == static_cast<int>(FieldType::BOOLEAN_TRUE) ||
+                 type == static_cast<int>(FieldType::BOOLEAN_FALSE),
+               "expected bool field, got " + field_type_string(static_cast<FieldType>(type)) +
```

Comment:
```suggestion
  auto const field_type = static_cast<FieldType>(type);
  CUDF_EXPECTS(field_type == FieldType::BOOLEAN_TRUE ||
                 field_type == FieldType::BOOLEAN_FALSE,
               "expected bool field, got " + field_type_string(field_type) +
```
Fewer casts

Reply:
In your version `field_type` could be an invalid `FieldType` value. I wanted to avoid having potentially invalid `FieldType` vars by always casting in the other direction. But now I see that I'm passing `type` to `field_type_string` either way, so might as well avoid some casts.